### PR TITLE
Add ability to set a temporary trace directory

### PR DESCRIPTION
**Pull request by roquo**
This commit adds the ability to set a trace-directory with `-o=<DIR>`
The `<DIR>` path is then the trace directory. The name of the directory can be choosen by the user and doesn't need to be `<application>-<x>`.
Latest Trace is not updated, because I think it's not a good way to add a latest-trace symlink to this "temporary" trace-dir.
Replay is only possible by given the path name.

I wanted to add this feature, to store traces in more than one directory, by not changing the environment variable `_RR_TRACE_DIR` every time. When the option is not set, every works as it worked before.

Trace directory does now not get reomoved at all. 

I would like to add the ability to compress the trace directory after recording to tar.gz or tar.bz2 to save memory space. Would this be possible? And then you can start replay with a compressed archive, which then gets uncompressed and replayed. Time should not matter much in this aspect, i would like to save memory space.

### CODE REVIEW

**Comment by rocallahan** [comment]
> I would like to add the ability to compress the trace directory after recording to tar.gz or tar.bz2 to save memory space. Would this be possible? And then you can start replay with a compressed archive, which then gets uncompressed and replayed. Time should not matter much in this aspect, i would like to save memory space.

The files in the trace are already compressed, other than the `cloned_data_` and `mmap_` files. If you use `btrfs` or some other filesystem that supports file/block cloning, and the trace directory is in the same filesystem as most of your application files, then the `cloned_data_` and `mmap_clone` files won't take up any real extra space. That is the preferred way to minimize space usage because it's also very fast. So I don't really want to add an option to record into/replay from a tar file. You can of course add wrappers around rr to do this.

**Comment by rocallahan** [comment]
Please squash these commits together into a single commit

**Comment by roquo** [comment]
done

File: CMakeLists.txt [hunk 1]
```diff
@@ -736,6 +736,7 @@ set(BASIC_TESTS
   mremap_grow_shared
   mremap_non_page_size
   mremap_overwrite
+  mremap_private_grow_under_map
```
Comment (rocallahan) [inline CMakeLists.txt]: Something's gone wrong here, this change (and others in this commit) are already on master.

**Pull request closed without merging**